source/common/router/delegating_route_impl.cc

```diff
@@ -0,0 +1,188 @@
+#include "common/router/delegating_route_impl.h"
+
+namespace Envoy {
+namespace Router {
+
+// Router::DelegatingRoute
+const DirectResponseEntry* DelegatingRoute::directResponseEntry() const {
+  return base_route_->directResponseEntry();
+}
+
+const RouteEntry* DelegatingRoute::routeEntry() const { return base_route_->routeEntry(); }
+
+const Decorator* DelegatingRoute::decorator() const { return base_route_->decorator(); }
+
+const RouteTracing* DelegatingRoute::tracingConfig() const { return base_route_->tracingConfig(); }
+
+const RouteSpecificFilterConfig* DelegatingRoute::perFilterConfig(const std::string& name) const {
+  return base_route_->perFilterConfig(name);
+}
+
+// Router:DelegatingRouteEntry
+void DelegatingRouteEntry::finalizeResponseHeaders(
+    Http::ResponseHeaderMap& headers, const StreamInfo::StreamInfo& stream_info) const {
+  return base_route_->routeEntry()->finalizeResponseHeaders(headers, stream_info);
+}
+
+Http::HeaderTransforms
+DelegatingRouteEntry::responseHeaderTransforms(const StreamInfo::StreamInfo& stream_info) const {
+  return base_route_->routeEntry()->responseHeaderTransforms(stream_info);
+}
+
+const std::string& DelegatingRouteEntry::clusterName() const {
+  return base_route_->routeEntry()->clusterName();
+}
+
+Http::Code DelegatingRouteEntry::clusterNotFoundResponseCode() const {
+  return base_route_->routeEntry()->clusterNotFoundResponseCode();
+}
+
+const CorsPolicy* DelegatingRouteEntry::corsPolicy() const {
+  return base_route_->routeEntry()->corsPolicy();
+}
+
+absl::optional<std::string>
+DelegatingRouteEntry::currentUrlPathAfterRewrite(const Http::RequestHeaderMap& headers) const {
+  return base_route_->routeEntry()->currentUrlPathAfterRewrite(headers);
+}
+
+void DelegatingRouteEntry::finalizeRequestHeaders(Http::RequestHeaderMap& headers,
+                                                  const StreamInfo::StreamInfo& stream_info,
+                                                  bool insert_envoy_original_path) const {
+  return base_route_->routeEntry()->finalizeRequestHeaders(headers, stream_info,
```

Comment:
I was going to check coverage on this but for some reason the successful CI runs only latched DCO.
Can you make sure we have unit tests for all the delegating functions in this PR?

Reply:
I added a new unit test `DelegatingRouteEntryAllCalls`  to add coverage for all routeEntry delegating methods (except typedMetadata, ran into a strange issue with its mock not having defaults set, added a comment about it).

  Is this what you were imagining with "unit tests for all the delegating functions"?
  
    Left other Router::Route stuff alone (e.g. `directResponseEntry`) since this PR doesn’t define a "DelegatingDirectResponseEntry" mechanism